docs_src/manuals/user_manual/execution_model.hrst

```diff
@@ -31,15 +31,105 @@ execution policy. A computation combines several multistages and executes one mu
   previous stages of the multistage have not already been applied to the indices in the same column with smaller ``k``.
 
 
+----------------------------------
+Extended Compute Domain of a Stage
+----------------------------------
+
+For a clear specification of access restrictions, we need to introduce the concept of extended compute domains.
+
+If a stage is accessing neighbor grid points of field `a`, the computation will consume halo points at the boundary of the compute domain.
+In case field `a` was written in a previous stage of the same multi-stage, the compute domain of the first stage needs to be extended, such
+that the *extended compute domain* of the stage covers the extent of the read access in the later stage.
+
 -------------------
 Access restrictions
 -------------------
 
-The execution model imposes restrictions on how accessors can be evaluated. The following restrictions apply:
+The execution model imposes restrictions on how accessors can be evaluated. These restrictions are not checked by GridTools,
+violating these rules results in undefined behavior.
+
+Definitions:
+
+ - *Writing/reading to/from an accessor* is short for *writing/reading to/from a field through an accessor*.
+   Note that the same field could be bound to different accessors, though this is not recommended.
+
+- *Previous/later statement* is short for *previous/later statement in the same stage or any previous/later stage*.
+
+
+The following rules apply to accesses in stages *within the same multi-stage*:
+
+ 1. A stage may write to an accessor, if there is no read from the same accessor in a previous statement.
+
+ 2. Accessors can be written and read in a later statement, if all reads of
+    this accessor are in stages where the compute domain is not extended.
```

Comment:
I think write-read in general is OK. Certain points might be evaluated twice in extended compute domain, but as long as it is only a write-read, there is no conflict: Assume we have two threads evaluating field X:
(1) T1: W(X)
(2) T2: W(X)
(3) T1: R(X)
(4) T2: R(X)

We also know that (1) -> (3) (meaning "1 happens before 3") and (2) -> (4). No matter how you change the order, the read always happens after at least one write and you get correct results.

You only get a problem if after the read, the field is written again.

(1) T1: R1(X)
(2) T2: R1(X)
(3) T1: W(X)
(4) T2: W(X)
(5) T1: R2(X)
(6) T2: R2(X)
1->3->5 and 2->4->6

Now this can happen in this order:
(1) T1: R1(X)
(3) T1: W(X)
(5) T1: R2(X)
(2) T2: R1(X) <--- ERROR!
(4) T2: W(X)
(6) T2: R2(X)

Same also happens for write-read-write.